src/gui/speciestab_sites.cpp

```diff
@@ -27,26 +24,25 @@ SpeciesSite *SpeciesTab::currentSite()
 
 void SpeciesTab::setCurrentSiteFromViewer()
 {
-    Locker refreshLocker(refreshLock_);
-
     // Get the currently-displayed site in the SiteViewer - if different from ours, change our controls to reflect it
-    SpeciesSite *displayedSite = ui_.SiteViewerWidget->siteViewer()->speciesSite();
+    auto *displayedSite = ui_.SiteViewerWidget->siteViewer()->speciesSite();
     if (!displayedSite)
         return;
 
-    // Sanity check that the displayed site actually exists in our species
-    if (std::find_if(species_->sites().begin(), species_->sites().end(),
-                     [displayedSite](const auto &p) { return &p == displayedSite; }) == species_->sites().end())
+    // Sanity check that the displayed site actually exists in our species, and get its index in the sites list
+    auto index = 0;
+    for (auto &site : species_->sites())
+    {
+        if (site.get() == displayedSite)
+            break;
+        ++index;
+    }
+    if (index >= species_->sites().size())
```

Comment:
```suggestion
    // Sanity check that the displayed site actually exists in our species, and get its index in the sites list
    auto it = std::find_if(species_->sites().begin(), species_->sites.end(),
        [displayedSite](const auto &site){ return site.get() == displayedSitel }));
    auto index = it - species_->sites().begin();
    if (it == species->sites().end())
```

This pulls us back to the finding logic while still retrieving the index value.